clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -307,9 +307,8 @@ public void updateFetchPositions(Set<TopicPartition> partitions) {
                         if (error == Errors.INVALID_TOPIC_EXCEPTION)
                             throw new InvalidTopicException("Topic '" + topic + "' is invalid");
                         else if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION)
-                            // if a requested topic is unknown, we just continue and let it be absent
-                            // in the returned map
-                            continue;
+                            throw new KafkaException("Topic '" + topic + "' may not exist or the user may not have " +
```

Comment:
I agree it is worth discussion. I just want to raise the concern that this may break user's code, e.g. if user doesn't catch exception and determines whether topic exists by looking at the map returned by the method.